### ISSUE TITLE
FNX3-17062 ⁃ Opening a sent tab crashes Fenix. [RuntimeException: Failed to call observer method]

### ISSUE THREAD

**Issue by agi90** [mozilla-mobile/fenix#3505]
https://sentry.prod.mozaws.net/operations/fenix/issues/5229222/

```
IllegalStateException: Display already acquired
    at org.mozilla.geckoview.GeckoSession.acquireDisplay(GeckoSession.java:7)
    at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:19)
    at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:5)
    at mozilla.components.browser.engine.gecko.GeckoEngineView.render(GeckoEngineView.kt:9)
    at mozilla.components.feature.session.EngineViewPresenter.renderSession$feature_session_release(EngineViewPresenter.kt:1)
...
(44 additional frame(s) were not displayed)

RuntimeException: Failed to call observer method
    at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:6)
    at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:2)
    at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:2)
    at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:3)
    at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:39)
...
(34 additional frame(s) were not displayed)

Failed to call observer method
```

**Comment by agi90** [comment]
STR
- Send tab from desktop firefox
- Manually Sync on Fenix (this might not be necessary)
- Open sent tab from notification pop up
- observe crash

**Comment by colintheshots** [comment]
Is this in the most recent nightly? Which version? Thanks @agi90!

**Comment by agi90** [comment]
Latest nightly for me (play store doesn't offer an update).
![Screenshot_20190615-131325_Firefox Preview](https://user-images.githubusercontent.com/4297388/59555934-3c46dc80-8faa-11e9-8df6-bfc2a0ce3353.jpg)

**Comment by agi90** [comment]
Note: you have to click on the pop-up widget to reproduce the crash (not sure every android version has it?)

![20190615_132413](https://user-images.githubusercontent.com/4297388/59556039-b461d200-8fab-11e9-9272-6f3b1180073b.jpg)

**Comment by colintheshots** [comment]
It crashes each time I open from the notification. When debugging this, we clearly try to open the same tab twice. That is exactly how we produced this crash trace in the past. I'm looking into a root cause.

**Comment by colintheshots** [comment]
If sending tabs is enabled again, we need to test this fix.

**Comment by softvision-miralobontiu** [comment]
It is not fixed yet. The app doesn't crash, but when tapping on the notification, nothing happens. 
I think #4283 is a dupe of the actual behavior.

[logcat.txt](https://github.com/mozilla-mobile/fenix/files/3430496/logcat.txt)

![20190725_Trim](https://user-images.githubusercontent.com/40694158/61859232-673d1e00-aed0-11e9-9584-59fcc63acb4f.gif)

**Comment by colintheshots** [comment]
It WAS fixed. Then someone stripped the fix from the code and caused other problems at the same time. I'll re-fix it.

**Comment by colintheshots** [comment]
With PR #4764, I think this should be working again. Please verify.

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on RC 1.3.0-rc.3, and on the latest Nightly 190819 (#12310612) with Huawei P9 Lite (Android 6).
